include/thread_pool/worker.hpp

```diff
@@ -165,10 +165,10 @@ class Worker final
 
 namespace detail
 {
-    inline size_t* thread_id()
+    inline std::atomic<size_t>& thread_id()
     {
-        static thread_local size_t tss_id = std::numeric_limits<size_t>::max();
-        return &tss_id;
+        static thread_local std::atomic<size_t> tss_id(std::numeric_limits<size_t>::max());
```

Comment:
why does a thread local need to be atomic?

Reply:
There is a race between when this gets assigned (at the start of `threadFunc`) and the first post that occurs (which can query for `tss_id`).

Reply:
My above comment is wrong, will update :)

Reply:
Fixed!